pandera/backends/pyspark/container.py

```diff
@@ -283,7 +283,7 @@ def collect_schema_components(
         schema_components = []
         for col_name, column in schema.columns.items():
             if (
-                column.required or col_name in check_obj
+                column.required or col_name in check_obj.columns
```

Comment:
Currently, it raises th `ValueError: Cannot convert column into bool: ...` error message.

Reply:
can you please elaborate specific condition for which it fails?

Reply:
Sure: a screenshot for the `pdb` session where the `col_name` and `check_obj` are displayed:
<img width="1511" alt="image" src="https://github.com/unionai-oss/pandera/assets/110418479/f7465ad5-49ee-41bd-96ae-da93ac02deb0">

Reply:
@filipeo2-mck is this a `PySpark` version related issue? What version do you have locally and what's specifified in the requirements?

Reply:
I have current `main` version locally, installed in editable mode.
It looks to be only [PySpark related issue](https://github.com/unionai-oss/pandera/blob/ceeae10f0fcca5f34de99d5e1c107ddacff51b73/pandera/backends/pyspark/container.py#L286).

BTW, in the same `container.py` we have [other usages](https://github.com/unionai-oss/pandera/blob/ceeae10f0fcca5f34de99d5e1c107ddacff51b73/pandera/backends/pyspark/container.py#L242C54-L242C54) of `check_obj.columns`.